code/core/src/preview-errors.ts

```diff
@@ -317,3 +318,22 @@ export class UnknownArgTypesError extends StorybookError {
     });
   }
 }
+
+export class UnsupportedViewportDimensionError extends StorybookError {
+  constructor(public data: { dimension: string; value: string; }) {
+    super({
+      category: Category.ADDON_VITEST,
+      code: 1,
+      // TODO: Add documentation about viewports support
+      // documentation: '',
+      message: dedent`
+        Encountered an unsupported value "${data.value}" when setting the viewport ${data.dimension} dimension.
+        
+        The Storybook plugin only supports values in the following units:
+        - px, vh, vw, em, rem and %.
+
+        You can either change the viewport for this story or use one of the supported units.
```

Comment:
```suggestion
        You can either change the viewport for this story to use one of the supported units
        or skip the test by adding `'!test'` to the story's tags per https://storybook.js.org/docs/writing-stories/tags
```

How about something like this?